P0009/P0009.bs

```diff
@@ -2364,32 +2316,51 @@ constexpr mapping get_mapping() const noexcept;
 constexpr bool is_unique() const noexcept;
 ```
 
-* *Returns:* `get_mapping().is_unique()`
+* *Returns:* `mapping().is_unique()`
 
 <br/>
 
 ```c++
 constexpr bool is_contiguous() const noexcept;
 ```
 
-* *Returns:* `get_mapping().is_contiguous()`
+* *Returns:* `mapping().is_contiguous()`
 
 <br/>
 
 ```c++
 constexpr bool is_strided() const noexcept;
 ```
 
-* *Returns:* `get_mapping().is_strided()`
+* *Returns:* `mapping().is_strided()`
 
 <br/>
 
 ```c++
 constexpr index_type stride(size_t r) const;
 ```
 
-* *Returns:* `get_mapping().stride(r)`
+* *Returns:* `mapping().stride(r)`
+
+<br/>
+
+```c++
+template<class... IndexType>
+static constexpr index_type required_span_size(IndexType... dynamic_extents) noexcept;
```

Comment:
We discussed this and were not in favor of replicating too much stuff in mdspan which exists in mapping. But it wasn't a super strong preference. If you like it to be in here that is fine.

Reply:
It's not just about replicating too much stuff. This won't work for non-unique, stateful mappings, so the operational semantics would have to somehow exclude those mappings, and the mapping would have to somehow advertise that its statefulness affects its required span size. I'm okay with putting it back in if those problems are solved, but this pull request doesn't solve those problems AFAICT.

Reply:
The `mapping` version requires construction of a mapping object and then function call on that object, a syntactically long two-step operation .  I believe this will be a frequent pattern and the shortcut is a desirable convenience.  The other single-step operations could be removed.